src/cargo/core/shell.rs

```diff
@@ -484,16 +508,17 @@ mod imp {
                 // resize the console correctly, but there's no reasonable way
                 // to detect which kind of terminal we are running in, or if
                 // GetConsoleScreenBufferInfo returns accurate information.
-                return Some(cmp::min(60, width));
+                return TtyWidth::Guess(cmp::min(60, width));
             }
-            None
+
+            TtyWidth::NoTty
         }
     }
 }
 
 #[cfg(windows)]
 fn default_err_erase_line(shell: &mut Shell) {
-    if let Some(max_width) = imp::stderr_width() {
+    if let TtyWidth::Known(max_width) = imp::stderr_width() {
```

Comment:
This also needs to handle the Guess case.

Reply:
Fixed, thanks!